src/Storages/StorageValues.cpp

```diff
@@ -32,6 +46,25 @@ Pipe StorageValues::read(
 {
     storage_snapshot->check(column_names);
 
+    if (!prepared_pipe.empty())
+    {
+        auto dag = std::make_shared<ActionsDAG>(prepared_pipe.getHeader().getColumnsWithTypeAndName());
+        ActionsDAG::NodeRawConstPtrs outputs;
+        outputs.reserve(column_names.size());
+        for (const auto & name : column_names)
+            outputs.push_back(dag->getOutputs()[prepared_pipe.getHeader().getPositionByName(name)]);
+
+        dag->getOutputs().swap( outputs);
+        auto expression = std::make_shared<ExpressionActions>(dag);
```

Comment:
Does this expression required for reordering columns in pipe output according to `column_names`?

Reply:
Yes, reordering and removing unneeded columns.